app/database/models/ms_schema/tasks_list.py

```diff
@@ -0,0 +1,210 @@
+    ) -> None:
+        """Adds a task to the list of tasks.
+        
+        Args:
+            description: A description of the task added.
+            created_at: Date on which the task is created.
+            is_done: Boolean specifying completion of the task.
+            completed_at: Date on which task is completed.
+        """
+
+        task = {
+            TasksFields.ID.value: self.next_task_id,
+            TasksFields.DESCRIPTION.value: description,
+            TasksFields.IS_DONE.value: is_done,
+            TasksFields.CREATED_AT.value: created_at,
+            TasksFields.COMPLETED_AT.value: completed_at,
+        }
+        self.next_task_id += 1
+        self.tasks = self.tasks + [task]
+
+    def delete_task(self, task_id: int) -> None:
+        """Deletes a task from the list of tasks.
+
+        Args:
+            task_id: Id of the task to be deleted.
+        """
+
+        new_list = list(
+            filter(lambda task: task[TasksFields.ID.value] != task_id, self.tasks)
+        )
+
+        self.tasks = new_list
+        self.save_to_db()
+
+    def update_task(
+        self,
+        task_id: int,
+        description: str = None,
+        is_done: bool = None,
+        completed_at: date = None,
+    ) -> None:
+        """Updates a task.
+        
+        Args:
+            task_id: Id of the task to be updated.
+            description: A description of the task.
+            created_at: Date on which the task is created.
+            is_done: Boolean specifying completion of the task.
+            completed_at: Date on which task is completed.
+        """
+
+        new_list = []
+        for task in self.tasks:
+            if task[TasksFields.ID.value] == task_id:
+                new_task = task.copy()
+                if description is not None:
+                    new_task[TasksFields.DESCRIPTION.value] = description
+
+                if is_done is not None:
+                    new_task[TasksFields.IS_DONE.value] = is_done
+
+                if completed_at is not None:
+                    new_task[TasksFields.COMPLETED_AT.value] = completed_at
+
+                new_list = new_list + [new_task]
+                continue
+
+            new_list = new_list + [task]
+
+        self.tasks = new_list
+        self.save_to_db()
+
+    def find_task_by_id(self, task_id: int):
+        """Returns the task that has the specified id.
+        
+        Args:
+            task_id: Id of the task.
+
+        Returns:
+            The task instance.
+        """
+        task = list(
+            filter(lambda task: task[TasksFields.ID.value] == task_id, self.tasks)
+        )
+        if len(task) == 0:
+            return None
+        return task[0]
+
+    def is_empty(self) -> bool:
+        """Checks if the list of tasks is empty.
+
+        Returns:
+            Boolean; True if the task is empty, False otherwise.
+        """
+
+        return len(self.tasks) == 0
```

Comment:
self.tasks

Reply:
return self.tasks would directly work. Please don't use len to check if its empty. Please fix everywhere in the code.

Reply:
not sure why this passed the checks on mentorship system. I didn't write this code, I copy pasted from the MS directly just so SQLAlchemy didn't throw tantrum on BIT schema

Reply:
will do the change on BIT